File: src/main/java/co/aplicared/jvm/juego/arbúcies/util/Compass.java
package co.aplicared.jvm.juego.arbúcies.util;

public enum Compass {
    NORTH(0, "North"),
    EAST(1, "East"),
    SOUTH(2, "South"),
    WEST(3, "West");

    private final int val;
    private final String description;

    Compass(int i, String description) {
        val = i;
        this.description = description;
    }

    public int value() {
        return val;
    }

    public String desc() {
        return description;
    }
}

File: src/main/kotlin/co/aplicared/jvm/juego/arbucies/util/Compass.kt
package co.aplicared.jvm.juego.arbucies.util

enum class Compass(private val `val`: Int, private val description: String) {
    NORTH(0, "North"),
    EAST(1, "East"),
    SOUTH(2, "South"),
    WEST(3, "West");

    fun value(): Int {
        return `val`
    }

    fun desc(): String {
        return description
    }

}